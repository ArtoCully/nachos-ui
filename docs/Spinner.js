import React from 'react'
import { View } from 'react-native'

import Spinner from '../src/Spinner'
import H4 from '../src/typography/H4'

// WEBPACK (SpinnerExample)
const SpinnerExample = () => {
  return (
    <View>
      <H4>Example:</H4>
      <View style={{ margin: 15 }}>
        <Spinner />
      </View>
    </View>
  )
}
// WEBPACK

const description = `
| Prop | Type | Default | Description |
|------|------|---------|-------------|
| \`duration\` | \`number\` | \`450\` | Specifies a duration of the animation |
| \`size\` | \`number\` | \`25\` | Specifies a size of the one spinner's circle  |
| \`color\` | \`string\` | \`#2f8cff\` | Specifies a color of spinner's circle |
| \`theme\` | \`object\` | \`defaultTheme\` | An object describing a style of the component. See more in Theme section |
`

SpinnerExample.styleguide = {
  ...SpinnerExample.styleguide,
  title: 'Spinner',
  description,
}

export default SpinnerExample
