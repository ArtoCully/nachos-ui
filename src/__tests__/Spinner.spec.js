import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import Spinner from '../Spinner'

it('should render correctly', () => {
  const component = renderer.create(<Spinner />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
