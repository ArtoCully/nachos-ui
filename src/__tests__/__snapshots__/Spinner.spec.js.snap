exports[`test should render correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }>
  <View
    style={
      Object {
        "backgroundColor": "#2f8cff",
        "borderRadius": 25,
        "height": 25,
        "transform": Array [
          Object {
            "scale": 0.4,
          },
        ],
        "width": 25,
      }
    } />
  <View
    style={
      Object {
        "backgroundColor": "#2f8cff",
        "borderRadius": 25,
        "height": 25,
        "transform": Array [
          Object {
            "scale": 0.4,
          },
        ],
        "width": 25,
      }
    } />
  <View
    style={
      Object {
        "backgroundColor": "#2f8cff",
        "borderRadius": 25,
        "height": 25,
        "transform": Array [
          Object {
            "scale": 0.4,
          },
        ],
        "width": 25,
      }
    } />
</View>
`;
